From 7cf3b0315dca73b83f4889b48fa44af85f90d8d0 Mon Sep 17 00:00:00 2001
From: Patrick Welche <prlw1@cam.ac.uk>
Date: Wed, 8 Nov 2017 22:52:19 +0000
Subject: [PATCH] python: in nodeWrap deal with name is None case

Should fix gtk-doc 1.26 build with itstool 2.0.4:

Error: Could not merge translations:
'NoneType' object is not subscriptable

https://bugzilla.gnome.org/show_bug.cgi?id=790071
---
 python/libxml.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/python/libxml.py b/python/libxml.py
index 2466cc9f..e53fb40d 100644
--- a/python/libxml.py
+++ b/python/libxml.py
@@ -573,6 +573,8 @@ class xmlCoreBreadthFirstItertor:
 def nodeWrap(o):
     # TODO try to cast to the most appropriate node class
     name = libxml2mod.type(o)
+    if name is None:
+        return xmlNode(_obj=o)
     if name == "element" or name == "text":
         return xmlNode(_obj=o)
     if name == "attribute":
-- 
2.15.0

